scripts/release.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import os
+import pipes
+import subprocess
+import traceback
+
+import click
+
+import ghutils
+from notifications import mail, mailgun, try_context
+
+logger = logging.getLogger("release")
+
+
+# release notes
+@click.command()
+@click.option(
+    "--script",
+    'scripts',
+    help="Execute these script(s) to generate release notes, may be specified"
+         " multiple times. After all scripts are executed, release notes must "
+         " have been written to $WORKSPACE/artifacts/release_notes",
+    multiple=True
+)
+@click.option(
+    "--file", "rnfile",
+    help="Read release notes from this file."
+)
+@click.option(
+    "--text",
+    help="Text of the release notes."
+)
+@click.option(
+    "--ref",
+    help="Ref that will be released. Release notes scripts compare from"
+         " --prev-version to --ref. May be omitted if --ref has already"
+         " been supplied to clone"
+)
+@click.option(
+    "--prev-version",
+    help="Last released version. Release notes scripts should compare "
+         "prev-version to ref"
+)
+@click.option(
+    "--clone-dir",
+    help="Root of the repo dir. May be omitted if clone was used earlier"
+         " in the chain."
+)
+def generate_release_notes(scripts, rnfile, text, ref, prev_version,
+                           clone_dir):
+    ctx_obj = click.get_current_context().obj
+    clone_dir = try_context(ctx_obj, clone_dir, "clone_dir", "clone_dir")
+    # cwd == $WORKSPACE
+    artifact_dir = "{cwd}/artifacts".format(cwd=os.getcwd())
+    dst_file = "{artifact_dir}/release_notes".format(artifact_dir=artifact_dir)
+    os.system("mkdir -p {}".format(artifact_dir))
+    if scripts:
+        ref = try_context(ctx_obj, ref, "ref", "rc_ref")
+        logger.debug(
+            "Generating release notes from scripts: {}".format(scripts)
+        )
+        sub_env = os.environ.copy()
+        sub_env["PREVIOUS_VERSION"] = prev_version
+        sub_env["VERSION"] = ref
+        script_work_dir = "{cwd}/{clone_dir}".format(
+            cwd=os.getcwd(),
+            clone_dir=clone_dir
+        )
+        logger.debug("script work dir: {}".format(script_work_dir))
+        for script in scripts:
+            try:
+                script_path = "{script_work_dir}/{script}".format(
+                    script_work_dir=script_work_dir,
+                    script=script
+                )
+                if os.path.exists(script_path):
```

Comment:
Along the lines of my other comment, the script(s) should take the release notes path as an argument or maybe just write to stdout so that we are specifying a specific location in less places. That allows us to change how this is all done without affecting the users.

Reply:
See https://github.com/rcbops/rpc-gating/pull/386/files#r141568148

I'll add the path as an option in the python implementation then pass in the path from the job.